File: src/gasing/division/division10.py
import time

a = 123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890
b = 10

# GASING Division
start = time.time()
for _ in range(10000000):
        hasil = int(str(a)[:-1])
end = time.time()
gasing_time = end - start
print("GASING Division 10:", gasing_time, "seconds")
print("GASING Result:", hasil)

# Computer Division
start = time.time()
for _ in range(10000000):
    hasil = a // b
end = time.time()
computer_time = end - start
print("General Division 10:", computer_time, "seconds")
print("Computer Result:", hasil)

print("\nGASING is", computer_time/gasing_time, "times faster")

File: src/gasing/division/division10.rs
use std::time::Instant;

fn main() {
    let a: i64 = 123456789;
    let b: i64 = 10;

    // GASING Division
    let start = Instant::now();
    let mut hasil: i64 = 0;
    for _ in 0..10000000 {
        hasil = a / 10; // In Rust, we'll use direct division instead of string manipulation
    }
    let gasing_time = start.elapsed().as_secs_f64();
    println!("GASING Division 10: {} seconds", gasing_time);

    // Computer Division
    let start = Instant::now();
    for _ in 0..10000000 {
        hasil = a / b;
    }
    let computer_time = start.elapsed().as_secs_f64();
    println!("General Division 10: {} seconds", computer_time);

    println!("\nGASING is {} times faster", computer_time/gasing_time);
}